src/service/dashboards/reports.rs

```diff
@@ -141,9 +141,9 @@ pub async fn save(
         // Supports only one tab for now
         let tab_id = &dashboard.tabs[0];
         tasks.push(async move {
-            let dashboard = db::dashboards::get(org_id, dash_id, folder).await?;
+            let maybe_dashboard = db::dashboards::get(org_id, dash_id, folder).await?;
             // Check if the tab_id exists
-            if let Some(dashboard) = dashboard.v3 {
+            if let Some(dashboard) = maybe_dashboard.and_then(|d| d.v3) {
                 let mut tab_found = false;
```

Comment:
I don't know much about reports so I don't know why this logic is only executed for V3 of dashboards and not any other version. But the changes here should preserve the existing behavior from before this PR.